File: modulo4/flutter_shop_app/lib/presentation/widgets/kpi_card.dart
// lib/presentation/widgets/kpi_card.dart

import 'package:flutter/material.dart';
import '../../theme/app_colors.dart';

class KpiCard extends StatelessWidget {
  final String       title;
  final String       value;
  final String?      subtitle;
  final IconData     icon;
  final Color        color;
  final bool         hasAlert;
  final VoidCallback? onTap;

  const KpiCard({
    super.key,
    required this.title,
    required this.value,
    required this.icon,
    this.subtitle,
    this.color    = AppColors.accent,
    this.hasAlert = false,
    this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    final tt = Theme.of(context).textTheme;

    Widget card = Container(
      padding:    const EdgeInsets.all(14),
      decoration: BoxDecoration(
        color:        AppColors.surface,
        borderRadius: BorderRadius.circular(16),
        border:       hasAlert
            ? Border.all(color: AppColors.error.withValues(alpha: 0.3))
            : Border.all(color: AppColors.border),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Container(
                width:  38, height: 38,
                decoration: BoxDecoration(
                  color:        color.withValues(alpha: 0.12),
                  borderRadius: BorderRadius.circular(10),
                ),
                child: Icon(icon, color: color, size: 20),
              ),
              if (hasAlert)
                const Text('⚠️', style: TextStyle(fontSize: 16)),
            ],
          ),
          const SizedBox(height: 12),
          Text(
            value,
            style: TextStyle(
              color:      AppColors.textPrimary,
              fontSize:   26,
              fontWeight: FontWeight.w800,
              height:     1,
            ),
          ),
          const SizedBox(height: 2),
          Text(title, style: tt.bodySmall),
          if (subtitle != null) ...[
            const SizedBox(height: 2),
            Text(
              subtitle!,
              style: TextStyle(
                color:    hasAlert ? AppColors.error : AppColors.textFaint,
                fontSize: 11,
              ),
            ),
          ],
        ],
      ),
    );

    return onTap != null
        ? GestureDetector(onTap: onTap, child: card)
        : card;
  }
}

File: modulo2/shopapp/app/src/main/java/com/shopapp/presentation/ui/admin/dashboard/KpiCard.kt
// presentation/ui/admin/dashboard/KpiCard.kt
package com.shopapp.presentation.ui.admin.dashboard

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.shopapp.theme.*

@Composable
fun KpiCard(
    title:     String,
    value:     String,
    subtitle:  String?     = null,
    icon:      ImageVector,
    color:     Color       = Accent,
    hasAlert:  Boolean     = false,
    onClick:   (() -> Unit)? = null,
    modifier:  Modifier    = Modifier,
) {
    val container: @Composable (@Composable () -> Unit) -> Unit = { content ->
        if (onClick != null) {
            Surface(
                onClick        = onClick,
                shape          = MaterialTheme.shapes.large,
                color          = Surface,
                tonalElevation = 0.dp,
                modifier       = modifier.fillMaxWidth(),
            ) { content() }
        } else {
            Surface(
                shape          = MaterialTheme.shapes.large,
                color          = Surface,
                tonalElevation = 0.dp,
                modifier       = modifier.fillMaxWidth(),
            ) { content() }
        }
    }

    container {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .then(
                    Modifier.background(
                        color  = color.copy(alpha = 0.06f),
                        shape  = MaterialTheme.shapes.large,
                    )
                )
                .padding(16.dp),
        ) {
            Row(
                modifier              = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment     = Alignment.Top,
            ) {
                Box(
                    modifier         = Modifier
                        .size(40.dp)
                        .background(color.copy(alpha = 0.15f), MaterialTheme.shapes.medium),
                    contentAlignment = Alignment.Center,
                ) {
                    Icon(icon, contentDescription = null, tint = color, modifier = Modifier.size(22.dp))
                }
                if (hasAlert) {
                    Text("⚠️", fontSize = 18.sp)
                }
            }

            Spacer(Modifier.height(12.dp))

            Text(
                text       = value,
                fontSize   = 28.sp,
                fontWeight = FontWeight.ExtraBold,
                color      = TextPrimary,
                lineHeight = 30.sp,
            )
            Text(
                text  = title,
                style = MaterialTheme.typography.bodySmall,
                color = TextSecondary,
                fontWeight = FontWeight.Medium,
            )
            if (subtitle != null) {
                Text(
                    text  = subtitle,
                    style = MaterialTheme.typography.bodySmall.copy(fontSize = 11.sp),
                    color = if (hasAlert) Error else TextFaint,
                )
            }
        }
    }
}